tessera-jaxrs/common-jaxrs/src/test/java/com/quorum/tessera/api/filter/IPWhitelistFilterTest.java

```diff
@@ -105,54 +103,56 @@ public void hostInWhitelistGetsAccepted() {
     }
 
     @Test
-    public void errorFilteringStopsFutureFilters() {
+    public void defaultConstructor() {
+        when(configService.isUseWhiteList()).thenReturn(Boolean.TRUE);
+        MockServiceLocator mockServiceLocator = (MockServiceLocator) ServiceLocator.create();
+        mockServiceLocator.setServices(Collections.singleton(configService));
 
+        assertThat(new IPWhitelistFilter()).isNotNull();
+    }
+
+    @Test
+    public void localhostIsWhiteListed() {
+
+        URI peer = URI.create("http://localhost:8080");
+        when(configService.getPeers()).thenReturn(singletonList(peer));
 
-        when(configService.isUseWhiteList()).thenReturn(true);
-        // show that one request goes through okay
         final HttpServletRequest request = mock(HttpServletRequest.class);
-        doReturn("whitelistedHost").when(request).getRemoteAddr();
+        doReturn("127.0.0.1").when(request).getRemoteAddr();
+
         filter.setHttpServletRequest(request);
+
         filter.filter(ctx);
+
         verify(request).getRemoteHost();
         verify(request).getRemoteAddr();
-
         verifyNoMoreInteractions(ctx);
+    }
 
-        // show the second one errors
-        final HttpServletRequest requestError = mock(HttpServletRequest.class);
-        doThrow(RuntimeException.class).when(requestError).getRemoteHost();
-        filter.setHttpServletRequest(requestError);
-        filter.filter(ctx);
+    @Test
+    public void localhostIPv6IsAlsoWhiteListed() {
```

Comment:
This test is already covered by localHostIPv6IsAlsoWhiteListed

Reply:
Updated & extra test removed